### ISSUE TITLE
whitelist.tf named preset support

### ISSUE THREAD

**Issue by Wiethoofd** [AnAkkk/TFTrue#2]
_tftrue_whitelist_id_ only allows for integers (at the moment) and then downloads the whitelist file from the following url: _whitelist.tf/custom_whitelist__**int**_.txt_
(any letter in this convar value will result in a non-working whitelist!)

If any _non-number_ is set as ID it most likely is a preset available from the menu or a server whitelist filename; which will always be up to date + changes when presets/IDs are updated for new seasons.

_etf2l_6v6_ => redirects on whitelist.tf to the newest whitelist for etf2l 6s (__s21_ at the moment)
_etf2l_6v6__**s19** => the S19 whitelist (if the preset exists, which it does in this case)

Ideally the following should happen when changing ID/Preset for whitelists:

| _tftrue_whitelist_id_ | Download URL | Whitelist (1) | Preset | Redirects (2) |
| --- | --- | --- | --- | --- |
| [1](http://whitelist.tf/1) | /custom_whitelist_**1**.txt | allow everything | _Yes_ | Yes |
| [1250](http://whitelist.tf/1250) | /custom_whitelist_**1250**.txt | _ID 1250_ | _No_ | No |
| [3627](http://whitelist.tf/3627) | /custom_whitelist_**3627**.txt | ETF2L 6s (S21) | _Yes_ | Yes |
| [3835](http://whitelist.tf/3835) | /custom_whitelist_**3835**.txt | Nations Cup '15 | _Yes_ | Yes |
| [allow_everything](http://whitelist.tf/allow_everything) | /**allow_everything**.txt | allow everything | Yes | No |
| [etf2l_6v6](http://whitelist.tf/etf2l_6v6) | /**etf2l_6v6**.txt | ETF2L 6s (S21) | **latest** | Yes |
| [etf2l_whitelist_6v6](http://whitelist.tf/etf2l_whitelist_6v6) | /**etf2l_whitelist_6v6**.txt | ETF2L 6s (S21) | **latest** | _Yes_ |
| [etf2l_hl](http://whitelist.tf/etf2l_hl) | /**etf2l_hl**.txt | Nations Cup '15 | **latest** | Yes |
| [etf2l_6v6_s21](http://whitelist.tf/etf2l_6v6_s21) | /**etf2l_6v6_s21**.txt | ETF2L 6s (S21) | 6s S21 | No |
| [etf2l_9v9_s8](http://whitelist.tf/etf2l_9v9_s8) | /**etf2l_9v9_s8**.txt | ETF2L HL (S8) | Yes | No |

1st column ID/preset is a link to the whitelist.tf overview
**(1)** On July 1st, 2015
**(2)** without .txt at the end of the url redirects occur (filenames will only redirect for non existing presets)

**TL;DR**
- IDs are unique for whitelists and don't change
- presets/filenames represent a whitelist ID (capitalization doesn't matter, all treated as lowercase)
  > presets can be changed to use different a ID (new season => different whitelist ID)
- ID or preset: the whitelist is always up to date respecting the item bans for that ID

**Comment by AnAkkk** [comment]
Done https://github.com/AnAkIn1/TFTrue/commit/f4bdceafe9af28d85b77d1c17c89486f1e1622d9

**Comment by AnAkkk** [comment]
Also https://github.com/AnAkIn1/TFTrue/commit/d8b2a5b30fb1fe29e292fdead2e8376098396669
